app/lib/report_helper.rb

```diff
@@ -427,10 +421,21 @@ def top_taxons_details(samples, background_id, num_results, sort_by_key, only_sp
       taxon_counts = res["taxon_counts"]
       sample_id = pr.sample_id
       tax_2d = convert_2d(taxon_counts)
+      cleanup_genus_ids!(tax_2d)
+      validate_names!(tax_2d)
+      cleanup_missing_genus_counts!(tax_2d)
+
+      if only_species
+        tax_2d.keep_if { |_tax_id, tax_info| tax_info['tax_level'] == TaxonCount::TAX_LEVEL_SPECIES }
+      else
+        tax_2d.keep_if { |_tax_id, tax_info| tax_info['tax_level'] == TaxonCount::TAX_LEVEL_GENUS }
+      end
+
       rows = []
       tax_2d.each do |_tax_id, tax_info|
         rows << tax_info
       end
+
       compute_aggregate_scores_v2!(rows)
```

Comment:
If you want to be completely be sync with the report page, you might want to go back to the original compute aggregate scores and move it above the genus/species selection.